tests/phpunit/integration/Core/Util/Method_Proxy_TraitTest.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Method_Proxy_TraitTest
+ *
+ * @package   Google\Site_Kit\Tests\Core\Util
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\Util;
+
+use Google\Site_Kit\Core\Util\Method_Proxy_Trait;
+use Google\Site_Kit\Tests\TestCase;
+
+/**
+ * Class Method_Proxy_TraitTest
+ *
+ * @group Util
+ */
+class Method_Proxy_TraitTest extends TestCase {
+
+	use Method_Proxy_Trait;
+
+	private $calls;
+
+	public function test_method_proxy() {
+		$this->calls = 0;
+		$callback    = $this->get_method_proxy( 'hidden_function' );
+
+		call_user_func( $callback );
+		call_user_func( $callback );
+
+		$this->assertEquals( $this->calls, 2 );
+	}
+
+	public function test_method_proxy_once() {
+		$this->calls = 0;
+		$callback    = $this->get_method_proxy_once( 'hidden_function' );
+
+		call_user_func( $callback );
+		call_user_func( $callback );
+
+		$this->assertEquals( $this->calls, 1 );
+	}
+
+	private function hidden_function() {
+		$this->calls++;
+	}
```

Comment:
You could use `MethodSpy` or use that as the base for a mock instance to use here instead of reimplementing that.

Reply:
I haven't reimplemented it, this case is different and I think don't think it is going to work with MethodSpy or a mock instance due to the nature of traits.

Reply:
I think it's possible but it looks more complicated than it's worth here for sure 👍